service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/twin/Twin.java

```diff
@@ -140,196 +154,37 @@ public Twin(String deviceId, String moduleId) throws IllegalArgumentException
         {
             throw new IllegalArgumentException("moduleId cannot be null or empty.");
         }
+
         this.deviceId = deviceId;
         this.moduleId = moduleId;
     }
 
-    /**
-     * Setter for twin tags.
-     *
-     * @param tags A set of tag key/value pairs.
-     * @throws IllegalArgumentException This exception is thrown if the set tags is {@code null}.
-     */
-    public void setTags(Set<Pair> tags) throws IllegalArgumentException
```

Comment:
Instead of setting, you can get the raw twincollection that allows you to put and putAll